Gems/Atom/RPI/Code/Source/RPI.Reflect/Material/MaterialPropertyValue.cpp

```diff
@@ -112,15 +113,37 @@ namespace AZ
             }
             else if (value.is<Data::AssetId>())
             {
-                result.m_value = Data::Asset<RPI::ImageAsset>(
-                    AZStd::any_cast<Data::AssetId>(value), azrtti_typeid<RPI::StreamingImageAsset>());
+                auto assetId = AZStd::any_cast<Data::AssetId>(value);
+                AZ::Data::AssetInfo assetInfo;
+                AZ::Data::AssetCatalogRequestBus::BroadcastResult(assetInfo, &AZ::Data::AssetCatalogRequests::GetAssetInfoById, assetId);
+                if (assetInfo.m_assetId.IsValid())
+                {
+                    result.m_value = Data::Asset<RPI::ImageAsset>(
+                        assetId,
+                        assetInfo.m_assetType,
+                        assetInfo.m_relativePath.c_str());
+                }
+                else
+                {
+                    result.m_value = Data::Asset<RPI::ImageAsset>(AZStd::any_cast<Data::AssetId>(value), azrtti_typeid<RPI::StreamingImageAsset>());
+                }
             }
             else if (value.is<Data::Asset<Data::AssetData>>())
             {
-                result.m_value = Data::Asset<RPI::ImageAsset>(
-                    AZStd::any_cast<Data::Asset<Data::AssetData>>(value).GetId(),
-                    azrtti_typeid<RPI::StreamingImageAsset>(),
-                    AZStd::any_cast<Data::Asset<Data::AssetData>>(value).GetHint());
+                auto assetId = AZStd::any_cast<Data::Asset<Data::AssetData>>(value).GetId();
+                AZ::Data::AssetInfo assetInfo;
+                AZ::Data::AssetCatalogRequestBus::BroadcastResult(assetInfo, &AZ::Data::AssetCatalogRequests::GetAssetInfoById, assetId);
+                if (assetInfo.m_assetId.IsValid())
+                {
+                    result.m_value = Data::Asset<RPI::ImageAsset>(
+                        assetId,
+                        assetInfo.m_assetType,
+                        AZStd::any_cast<Data::Asset<Data::AssetData>>(value).GetHint());
+                }
+                else
+                {
+                    result.m_value = AZStd::any_cast<Data::Asset<Data::AssetData>>(value);
+                }
```

Comment:
Do the any_cast once at the top of the block then reuse the asset or assetId.

Shouldn't the else case continue to convert to a streaming image asset?